crates/orchestrator/src/tests/jobs/mod.rs

```diff
@@ -501,3 +507,90 @@ fn build_job_item_by_type_and_status(job_type: JobType, job_status: JobStatus, i
         version: 0,
     }
 }
+
+#[rstest]
+#[case(JobType::SnosRun, JobStatus::Failed)]
+#[tokio::test]
+async fn handle_job_failure_with_failed_job_status_works(#[case] job_type: JobType, #[case] job_status: JobStatus) {
+    TestConfigBuilder::new().build().await;
+    let config = config().await;
+    let database_client = config.database();
+    let internal_id = 1;
+
+    // create a job, with already available "last_job_status"
+    let mut job_expected = build_job_item(job_type.clone(), job_status.clone(), internal_id);
+    let mut job_metadata = job_expected.metadata.clone();
+    job_metadata.insert("last_job_status".to_string(), JobStatus::PendingVerification.to_string());
+    job_expected.metadata = job_metadata.clone();
+
+    let job_id = job_expected.id;
+
+    // feeding the job to DB
+    database_client.create_job(job_expected.clone()).await.unwrap();
+
+    // calling handle_job_failure
+    handle_job_failure(job_id).await.expect("handle_job_failure failed to run");
+
+    let job_fetched = config.database().get_job_by_id(job_id).await.expect("Unable to fetch Job Data").unwrap();
+
+    assert_eq!(job_fetched, job_expected);
+}
+
+#[rstest]
+#[case::pending_verification(JobType::SnosRun, JobStatus::PendingVerification)]
+#[case::verification_timeout(JobType::SnosRun, JobStatus::VerificationTimeout)]
```

Comment:
why are the other statuses not covered ?

Reply:
We initially covered all the statuses as shown [here](https://github.com/madara-alliance/madara-orchestrator/blob/8db6859f7b53e4e205e2ec0276fbb309a579197b/crates/orchestrator/src/tests/jobs/mod.rs#L545), but it felt redundant to test all.

What do you suggest @EvolveArt  ?

Reply:
well even if it feels redundant I think it's important to have them, you want to avoid having jobs at an unexpected state in the DLQ

Reply:
Valid, Implemented